lib/AST/LifetimeDependence.cpp

```diff
@@ -504,20 +504,18 @@ class LifetimeDependenceChecker {
   }
 
   bool isCompatibleWithOwnership(ParsedLifetimeDependenceKind kind, Type type,
-                                 ValueOwnership ownership,
+                                 ValueOwnership loweredOwnership,
                                  bool isInterfaceFile = false) const {
     if (kind == ParsedLifetimeDependenceKind::Inherit) {
       return true;
     }
-    // Lifetime dependence always propagates through temporary BitwiseCopyable
-    // values, even if the dependence is scoped.
-    if (isBitwiseCopyable(type, ctx)) {
-      return true;
-    }
-    auto loweredOwnership = ownership != ValueOwnership::Default
-      ? ownership : getLoweredOwnership(afd);
-
     if (kind == ParsedLifetimeDependenceKind::Borrow) {
+      // An owned/consumed BitwiseCopyable value can be effectively borrowed
+      // because its lifetime can be indefinitely extended.
+      if (loweredOwnership == ValueOwnership::Owned
+          && isBitwiseCopyable(type, ctx)) {
```

Comment:
Shouldn't the same rule apply for `@_lifetime(&)` ? In which case we will need a similar check when `kind == ParsedLifetimeDependenceKind::Inout`

Reply:
```
struct C {}

struct NE : ~Escapable {
  @_lifetime(immortal)
  init() {}
}

@_lifetime(&c)
func foo(_ c: consuming C) -> NE {
  NE()
}
```

Reply:
okay, makes sense!